app/[locale]/job/edit/[idx]/page.test.tsx

```diff
@@ -1,51 +1,62 @@
-import { afterEach, beforeEach, describe, expect, it } from 'vitest'
+import { afterEach, beforeAll, beforeEach, describe, expect, it } from 'vitest'
 import { cleanup, fireEvent, render, screen, waitFor } from '@testing-library/react'
 import Page from './page'
 import { vi } from 'vitest'
 import mockRouter from 'next-router-mock'
 import TestWrapper from '@/app/TestWrapper'
 import { EnhancedStore } from '@reduxjs/toolkit/react'
 import { makeStore } from '@/lib/store'
-import { JobItem, addJob } from '@/lib/features/job/jobSlice'
+import { SetJobPayload, addJob } from '@/lib/features/job/jobSlice'
+import { createUuid } from '@/lib/store'
 
 describe('Edit Income Item Page', async () => {
     let store: EnhancedStore
-    const item1: JobItem = {
-        description: 'desc1',
-        business: 'business!',
-        taxesFiled: false,
-        payments: []
+    const id = createUuid()
+    const item1: SetJobPayload = {
+        id: id,
+        item: {
+            description: 'desc1',
+            business: 'business!',
+            taxesFiled: false
+        }
     }
-    beforeEach(() => {
+    beforeAll(() => {
```

Comment:
here i'm trying to separate the things that only need to happen once from the things that should happen before each test.